mutiny-core/src/proxy.rs

```diff
@@ -46,6 +46,31 @@ impl WsProxy {
             .map_err(|_| MutinyError::ConnectionFailed)?,
         };
 
+        // wait for connected status or time out at 30s
+        let mut retries = 30;
+        while retries > 0 {
+            match ws.state() {
+                State::Open => break,
+                State::Closed => break,
+                _ => {
+                    sleep(1_000).await;
+                    retries -= 1;
+                }
+            }
+        }
+
+        match ws.state() {
+            State::Open => {}
+            _ => return Err(MutinyError::ConnectionFailed),
+        }
+
+        // TODO wait until we get an OK response from websocket.
+        // A connection to the proxy for connections just means that
+        // it just connected to the proxy. It does not mean the proxy
+        // successfully connected out to the other end. They may be
+        // offline and shortly cut off from the WS but that happens
+        // outside of the connect flow. This will falsely return success.
```

Comment:
I will do this in a follow up when I have some more time. Essentially I need to update the ln-websocket-proxy and add a new `Command` type that returns `OK` once it successfully connected out to the other peer.